### PR TITLE
Disable Create collection button only when APO not open and not openable

### LINKED ISSUE #4410: "Create collection" button/modal/function should automatically handle APO versioning
Argo users should be able to use the "Create collection" function without having to open an APO first.
Screenshot 2024-04-12 at 2 28 34 PM

But the function updates the APO Cocina because it adds the new collection druid to the `collectionsForRegistration` list in the administrative metadata. Under the new approach to versioning, the APO object will need to be opened for this change to take place.

Request:

- Create collection button remains active even when object is locked
- Create collection modal allows user to create collection as is done now
- Submitting the modal functions the same way to the user
- Somewhere in that sequence, either while the user is editing or after they submit the form, the APO versioning is taken care of on the system side
- Version description should be "Created new collection: ${collection_druid}"

If it's not possible to insert the collection druid into the version description, just go with "Created new collection".

### PR DESCRIPTION
**Pull request by lwrubel**
# Why was this change made?
Resolves #4410 to adjust Create Collection button.

# How was this change tested?
Unit and testing on QA. Passed the `apo_creation_spec` and `collection_create_spec` integration tests. Will require PO testing.

### CODE REVIEW

File: app/controllers/collections_controller.rb [hunk 1]
```diff
@@ -35,6 +39,8 @@ def create
       )
     )
     Repository.store(updated_cocina_admin_policy)
+    # Close the APO version and reindex
+    version_service.close(description: "Created new collection: #{collection_druid}")
```
Comment (justinlittman) [inline app/controllers/collections_controller.rb:43]: You should be able to skip the description on close.

Comment (justinlittman) [reply app/controllers/collections_controller.rb:43]: I wouldn't overwrite an existing version description, but I don't think it really matters much.

**Review by justinlittman — APPROVED** [review]
Approve pending consideration of comment.

File: app/controllers/collections_controller.rb (hunk 1 above)
Comment (lwrubel) [reply app/controllers/collections_controller.rb]: Fixed to not overwrite an existing version description.

**Pull request merged**